src/auxinfo/info.rs

```diff
@@ -6,39 +6,97 @@
 // License, Version 2.0 found in the LICENSE-APACHE file in the root directory
 // of this source tree.
 
+use std::fmt::Debug;
+
 use crate::{
-    errors::Result,
+    errors::{InternalError, Result},
     paillier::{DecryptionKey, EncryptionKey},
     ring_pedersen::VerifiedRingPedersen,
+    ParticipantIdentifier,
 };
+use k256::elliptic_curve::zeroize::ZeroizeOnDrop;
 use libpaillier::unknown_order::BigNumber;
 use serde::{Deserialize, Serialize};
+use tracing::error;
 
 /// The private key corresponding to a given Participant's [`AuxInfoPublic`].
-#[derive(Serialize, Deserialize)]
+#[derive(Serialize, Deserialize, ZeroizeOnDrop)]
 pub(crate) struct AuxInfoPrivate {
-    pub(crate) sk: DecryptionKey,
+    decryption_key: DecryptionKey,
+}
+
+impl AuxInfoPrivate {
+    #[cfg(test)]
+    pub fn encryption_key(&self) -> EncryptionKey {
+        self.decryption_key.encryption_key()
+    }
+
+    pub fn decryption_key(&self) -> &DecryptionKey {
+        &self.decryption_key
+    }
+}
+
+impl From<DecryptionKey> for AuxInfoPrivate {
+    fn from(decryption_key: DecryptionKey) -> Self {
+        Self { decryption_key }
+    }
+}
+
+impl Debug for AuxInfoPrivate {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("AuxInfoPrivate")
+            .field("decryption key", &"[redacted]")
+            .finish()
+    }
 }
 
 /// The public Auxilary Information corresponding to a given Participant.
-#[derive(Serialize, Deserialize, Clone)]
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq)]
```

Comment:
The parameters are all IDs and public keys so it should be fine to log. Thanks for checking!

Reply:
Although honestly we might want to not log them because it's going to make the logs impossible to read. All these key types are massive. Maybe we should make a long-term issue to log these things in some condensed or hashed format.

Reply:
Haha. Is there any good reason for the `Debug` implementation to  print out the entire key types?

Reply:
Eh, it seems useful to have some way to distinguish between different keys in the logs, but that could be done without the entire key. If we're getting truly weird cryptographic behavior it might be useful to know but I don't really expect that to be the primary kind of errors we're debugging with logs.